### PR TITLE
Sound service

### PR DESCRIPTION
**Pull request by sambcox**
## Query/Mutation Effected

Sound Card Query

## Describe Changes

Adds the ability to query sound cards by category with the ability to provide a limit, with the default limit being set to 8. Adds the sound service and facade to be able to return links to the sounds themselves. Adds testing for all of this, including sad paths and edge cases.

## Requested Reviewer(s)

@this-is-joeking @Bontgoy @CassandraGoose

### CODE REVIEW

File: app/graphql/types/sound_card_type.rb [hunk 1]
```diff
@@ -5,7 +5,18 @@ class SoundCardType < Types::BaseObject
...
+    field :link, String, null: false
+    field :wrong_answers, [String], null: false
     field :created_at, GraphQL::Types::ISO8601DateTime, null: false
     field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
+
+    def link
+      SoundFacade.get_link(object.id)
+    end
+  
+
```
Comment (CassandraGoose) [inline app/graphql/types/sound_card_type.rb:17]: possible extra blank line?

**Review by CassandraGoose — COMMENTED** [review]
I'm not a BE/Rails/Graphql person, so I am unable to give much helpful feedback. 

One question I have is that I see 'figaro' has been included as a gem. Is that being used anywhere? (Perhaps this is just an env or config related item?)

From what I can tell, a request can now be made to get data for a specific sound (via a parameter of id). It looks like there were some updates to the fields so that instead of simply 'sound_cards', the entry point is now 'sound_cards_by_category'. I'm I on the right track with that? If so, I think it makes total sense to include that functionality. 

There also appears to be the functionality to create sound cards from a CSV for testing. Is this purely for testing? I'm afraid I'm not familiar enough with this ecosystem to understand the pros and cons of having tests dependent on CSV data - it's not really something we usually do in Node/JS land (at least not that I've seen) but if it works, it works. 

The tests look reasonable and the implementation code looks reasonable. Again, I'm not a Rails/BE/Graphql dev, so this is purely just surface level. 

I'm going to let you all 'approve' or 'request changes' instead of me, since y'all know more! From my end, seems solid.

**Comment by sambcox** [comment]
> I'm not a BE/Rails/Graphql person, so I am unable to give much helpful feedback.
> 
> One question I have is that I see 'figaro' has been included as a gem. Is that being used anywhere? (Perhaps this is just an env or config related item?)

You're correct here! It is an env item.


> 
> From what I can tell, a request can now be made to get data for a specific sound (via a parameter of id). It looks like there were some updates to the fields so that instead of simply 'sound_cards', the entry point is now 'sound_cards_by_category'. I'm I on the right track with that? If so, I think it makes total sense to include that functionality.

You are on the right track! It is more pointed and specific.


> 
> There also appears to be the functionality to create sound cards from a CSV for testing. Is this purely for testing? I'm afraid I'm not familiar enough with this ecosystem to understand the pros and cons of having tests dependent on CSV data - it's not really something we usually do in Node/JS land (at least not that I've seen) but if it works, it works.

This was purely to get data that was scraped by the whole team into the database, as that is held in a Google Sheet
> 
> The tests look reasonable and the implementation code looks reasonable. Again, I'm not a Rails/BE/Graphql dev, so this is purely just surface level.
> 
> I'm going to let you all 'approve' or 'request changes' instead of me, since y'all know more! From my end, seems solid.

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -7,15 +7,23 @@ class QueryType < Types::BaseObject
     # Add root-level fields here.
     # They will be entry points for queries on your schema.
     field :leaderboards, [Types::LeaderboardType], null: false
-    field :sound_cards, [Types::SoundCardType], null: false
+    field :sound_cards_by_category, [Types::SoundCardType], null: false do
+      argument :category, String, required: true
+      argument :limit, Integer, required: true, default_value: 8
```
Comment (this-is-joeking) [inline app/graphql/types/query_type.rb:12]: Like this default value 💯

File: app/graphql/types/query_type.rb [hunk 3]
```diff
@@ -7,15 +7,23 @@ class QueryType < Types::BaseObject
...
     def leaderboards
       Leaderboard.order(score: :desc)
     end
 
-    def sound_cards
-      SoundCard.all
+    def sound_cards_by_category(category:, limit:)
+      limit = 24 if limit > 24
+      unless category.downcase.include?('misc')
+        SoundCard.where("lower(category) = ?", category.downcase).shuffle.shuffle.first(limit)
```
Comment (this-is-joeking) [inline app/graphql/types/query_type.rb:23]: Case insensitivity 😏

**Review by this-is-joeking — APPROVED** [review]
Great work on this

**Pull request merged**